.vscode/launch.json

```diff
@@ -90,7 +90,9 @@
             "type": "cortex-debug",
             "request": "launch",
             "cwd": "${workspaceRoot}/examples/${input:mbedApp}/mbed",
+            "armToolchainPath": "${workspaceRoot}/.environment/cipd/pigweed/bin/",
```

Comment:
vscode dockerfile uses PW_ENVIRONMENT_ROOT: https://github.com/project-chip/connectedhomeip/blob/master/integrations/docker/images/chip-build-vscode/Dockerfile#L66

is there some method to say "use environment root, if that does not exist use the .environment?

Or are we assuming that when we debug we are not in  the vscode image anywan and people do not set this environment variable on their workstation?

Reply:
As far as I know, unfortunately,  there is no possibility to use conditions in the launch task. Of course, we can use PW_ENVIRONMENT_ROOT but then we assume that the VScode docker image is used. I think is more flexible to assume that we always need to bootstrap the Pigweed environment (which is necessary to build Mbed OS examples) and create the .environment directory. For example, if we want to use the Mbed OS docker image directly.

Reply:
Back to the issue. However, using a PW_ENVIRONMENT_ROOT variable will be a better solution because, I didn't notice that, but when we for vscode container the .environment directory is not created, we setup Pigweed inside PW_ENVIRONMENT_ROOT.  Assuming that we will only flash and debug Mbed applications in the VScode or Mbed container using PW_ENVIRONMENT_ROOT seems a more correct approach. I also added PW_ENVIRONMENT_ROOT to env for Mbed docker image to meet this assumption.